Mathlib/Analysis/CStarAlgebra/GNSConstruction/Defs.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2025 Gregory Wickham. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gregory Wickham
+-/
+module
+
+public import Mathlib.Analysis.CStarAlgebra.PositiveLinearMap
+public import Mathlib.Analysis.InnerProductSpace.Completion
+
+/-!
+# Definitions of structures for the GNS (Gelfand-Naimark-Segal) construction
+
+In this file we do the "construction" part of the GNS construction. We define a Hilbert space
+as the completion of a quotient of A after we mod out by an appropriately constructed subspace.
+The inner product is induced by a fixed positive linear functional `f`.
+
+Most of the structures, theorems, and definitions in this file should not be referenced directly,
+but they are described below for the sake of clarity and completeness.
+
+## Main results
+
+- `f.GNS` : a type synonym of `A` that "forgets" the norm of `A` and bundles in a fixed
+  linear functional `f` so that we can construct an inner product and inner product-induced norm.
+- `f.GNS_HilbertSpace` : the Hilbert space that we construct as the completion of `f.GNS_Quotient`.
+
+## References
+
+Most of this work follows from private course notes prepared by Professor Konrad Aguilar at Pomona
+College.
+
+For another, similar approach, see "A Primer on Spectral Theory" by Bernard Aupetit, the other main
+refence used here.
+-/
+
+@[expose] public section
+open scoped ComplexOrder
+
+variable {A : Type*} [NonUnitalCStarAlgebra A] [PartialOrder A] [StarOrderedRing A] (f : A →ₚ[ℂ] ℂ)
+
+namespace PositiveLinearMap
+/-- The GNS space on a non-unital C⋆-algebra with a positive linear functional. This erases the norm
+on `A`, while remainaing structurally equivalent via the `LinearEquivalence`, `toGNS`.
+-/
+def GNS (_f : A →ₚ[ℂ] ℂ) := A
```

Comment:
This lets you make the variable name `f` instead of `_f`.
```suggestion

set_option linter.unusedVariables false in
/-- The GNS space on a non-unital C⋆-algebra with a positive linear functional. This erases the norm
on `A`, while remainaing structurally equivalent via the `LinearEquivalence`, `toGNS`.
-/
@[nolint unusedArguments]
def GNS (f : A →ₚ[ℂ] ℂ) := A

```